lib/src/test/java/io/ably/lib/test/realtime/RealtimeAuthTest.java

```diff
@@ -74,6 +75,44 @@ public void auth_client_match_tokendetails_null_clientId() {
         }
     }
 
+    /**
+     * RSA4d:  If a request by a realtime client to an authUrl results in an HTTP 403 response,
+     * or any of an authUrl request, an authCallback, or a request to Ably to exchange
+     * a TokenRequest for a TokenDetails result in an ErrorInfo with statusCode 403,
+     * as part of an attempt by the realtime client to authenticate, then the client library
```

Comment:
Shouldn't this be testing that the 80019 code is also returned?

Reply:
I believe it does now. On the line 127,128 I'm checking error status code and error code as well. Also in `ConnectionStateListener` where I awaiting `ConnectionEven.failed`. Or do you have something else in mind?

Reply:
no, lgtm, thanks